trestle/core/commands/author/ssp.py

```diff
@@ -0,0 +1,364 @@
+                version to be printed out in the section header of the markdown.
+            yaml_header: The dictionary corresponding to the desired contents of the yaml header at the
+                top of each markdown file.
+        Returns:
+            0 on success, 1 otherwise
+
+        """
+        logging.debug(
+            f'Generate ssp in {md_path} from catalog {catalog.metadata.title}, profile {profile.metadata.title}'
+        )
+        # create the directory in which to write the control markdown files
+        md_path.mkdir(exist_ok=True, parents=True)
+
+        # build a convenience dictionary to access control handles by name
+        control_dict: Dict[str, ControlHandle] = {}
+        for group in catalog.groups:
+            for control in group.controls:
+                control_handle = ControlHandle(group.id, group.title, control)
+                control_dict = self._get_controls(control_handle, control_dict)
+
+        # get list of control_ids needed by profile
+        control_ids: List[str] = []
+        for _import in profile.imports:
+            for include_control in _import.include_controls:
+                control_ids.extend(include_control.with_ids)
+
+        # get list of group id's and associated controls
+        needed_group_ids: Set[str] = set()
+        needed_controls: List[ControlHandle] = []
+        for control_id in control_ids:
+            control_handle = control_dict[control_id]
+            needed_group_ids.add(control_handle.group_id)
+            needed_controls.append(control_handle)
+
+        # make the directories for each group
+        for group_id in needed_group_ids:
+            (md_path / group_id).mkdir(exist_ok=True)
+
+        # assign values to class members for use when writing out the controls
+        self._param_dict = profile.modify.set_parameters
+        self._alters = profile.modify.alters
+        self._yaml_header = yaml_header
+        self._sections = sections
+
+        # write out the controls
+        for control_handle in needed_controls:
+            out_path = md_path / control_handle.group_id
+            self._write_control(out_path, control_handle)
+
+        return 0
+
+    def _get_implementations(self, control_file: pathlib.Path) -> List[ossp.ImplementedRequirement]:
+        control_id = control_file.stem
+        parse_tree = MarkdownValidator.load_markdown_parsetree(control_file)
+        tree = parse_tree[0] if len(parse_tree) == 1 else parse_tree[1]
+        imp_reqs: list[ossp.ImplementedRequirement] = []
+        ii = 0
+        while ii < len(tree):
+            item = tree[ii]
+            if item['type'] == 'thematic_break':
+                section: str = tree[ii + 1]['children'][0]['text']
+                if not section.startswith('Part '):
+                    break
+                text = tree[ii + 2]['children'][0]['text']
+                by_comp: ossp.ByComponent = gens.generate_sample_model(ossp.ByComponent)
```

Comment:
I'm not sure we want to do this by component - can discuss.

Reply:
There will be many related changes needed in this section of code for 1.0.0 so I would prefer to defer to after 1.0.0

Reply:
Okay